ush/ufsda/r2d2.py

```diff
@@ -2,47 +2,87 @@
 from solo.configuration import Configuration
 from solo.date import date_sequence, Hour
 
+fv3files = ['fv_core.res', 'fv_srf_wnd.res', 'fv_tracer.res', 'phy_data', 'sfc_data']
+
 
 def store(config):
     times = date_sequence(config.start, config.end, config.step)
-    obs_types = config.obs_types
-    provider = config.provider
+    obs_types = config.get('obs_types', ['dummy'])
+    provider = config.get('provider', None)
     experiment = config.experiment
     database = config.database
     type = config.type
     file_type = config.get('file_type', None)
     source_dir = config.source_dir
     source_file_fmt = config.source_file_fmt
-    step = config.step
+    step = config.get('step', 'PT6H')
+    fcst_steps = config.get('forecast_steps', ['PT6H'])
     dump = config.get('dump', 'gdas')
+    resolution = config.get('resolution', 'c768')
+    model = config.get('model', 'gfs')
+    user_date_format = config.get('user_date_format', '%Y%m%d.%H%M%S')
+    fc_date_rendering = config.get('fc_date_rendering', 'analysis')
+    tiles = config.get('tiles', [1, 2, 3, 4, 5, 6])
+    file_type_list = config.get('file_type_list', fv3files)
 
     for time in times:
         year = Hour(time).format('%Y')
         month = Hour(time).format('%m')
         day = Hour(time).format('%d')
         hour = Hour(time).format('%H')
-        for obs_type in obs_types:
-            if type == 'bc':
+        if type in ['bc', 'ob']:
+            for obs_type in obs_types:
+                if type == 'bc':
```

Comment:
With several lines being identical between the if and else blocks, I wonder if we collapse this into a single `r2d2.store`.  Of course, this simply moves the if/else to where we populate the arguments sent to `rt2d.store`

Reply:
You made me think of a possible alternative solution, something in python called `**kwargs`. I bet we can use this to remove the if-statements entirely.

Reply:
the latest commit greatly reduces the amount of code in if-else blocks and still achieves the same result. Thanks for pointing me in the right direction :-)